### ISSUE TITLE
Update Code to Conform to Google JavaScript Style guide

### ISSUE THREAD

**Issue by dataist** [sims226/rivets#2]
Right now all of your code is at global scope (generally bad). I'd like you to refactor the code into javascript functions and keep an eye on your styling. 

The google style guide is a good place to start on writing well structured, easy to read, and safe code. You can find it [here](https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations). 

Wrap all your changes in a PR and I'll drop more comments there.

**Comment by dataist** [comment]
Note that the use of `var` is heavily discouraged because of it's global scope; Google actually disallows its use entirely with a preference for `const` or `let` where appropriate.

**Comment by sims226** [comment]
Okay, cool, I'll check these out.

I'm only vaguely familiar with `const` and `let` and the other ES2015 things. It's part of my problem with JavaScript: even when something works, I'm never quite sure if I'm doing it right. Not that I've written something that works yet...

On a similar note, I'm basically familiar with scopes and closures (as in I can pass a tutorial on it) but I'm not sure _why_ they matter. I know there's something about memory and something about technical debt, but it's part of that fundamental level of programming that is very important to me but not really in the _(pun alert)_ scope of a bootcamp.

Finally, at risk of rambling, there's a coin-flip's chance I'm about to take a part-time job as a pastry chef at Neidlov's in an effort to finance my new-and-improved code study as well as do something I consider honest, interesting work for a fair (if minimal) wage.

Finally finally, thanks for your help here. There's a 100% chance I'll be an imperfect student, but I'm determined to put in the work this time; just have to arrange life in a very very different (hopefully better) way.

**Comment by sims226** [comment]
Pushed new commits that refactor into smaller functions and abide Google's guidelines as best I can understand. Don't want to mark this complete until you've had a chance to look it over. I'm trying to read (not skim) through Twitter's API documentation, and I'm going to see if I can just study up on APIs in general.

**Comment by dataist** [comment]
all your changes should be commits on a feature branch

before you start new work run: 
1. `git branch some-new-feature-branch-name`
2. `git checkout some-new-feature-branch-name`
3. make some codes
4. add/commit changes (locally)
5. `git push` (to github)
6.  use github's PR functionality to create a request to merge `some-new-feature-branch-name` into `master`

let me know if any of that doesnt make sense

**Comment by sims226** [comment]
No, this makes sense; in fact, it falls into the LOOONG list of things that I've encountered, basically understood, but never actually put them into practice.

I'll branch for my next/following commits.

**Comment by sims226** [comment]
Are you familiar with this outfit?

http://exercism.io

Programming challenges, all done from CLI. The first JS lesson is "Hello, World!" but it immediately addresses test-driven stuff.

I admit to being easily distracted by new learning approaches, but I've hit a point of diminishing returns with the Code Schools and Treehouses of the world, looking for something to bridge the gap between "quiz passing" and "actually making shit with the tools you'll use to make shit."

Anyhoo, feedback appreciated.

**Comment by dataist** [comment]
No, but it looks cool. I might try them out next time I try and learn a
language. When you do these what do you find easiest and hardest about them?

**Comment by sims226** [comment]
I don't think the challenges in Exercism are uniquely difficult (disclaimer: all of programming makes me feel somewhat like I have a learning disability). Rather I like that it's fetched/submitted via the command line, and the challenge requires 'real' code, typed in a text editor, such that it will pass a series of Jasmine tests in the same exercise folder.

I use them sort of like a treadmill: hop over, break a sweat, occasionally feel good about it, move on to next challenge...
